lib/pure/collections/tables.nim

```diff
@@ -158,6 +158,9 @@ template getOrDefaultImpl(t, key): untyped =
   var index = rawGet(t, key, hc)
   if index >= 0: result = t.data[index].val
 
+template getOrDefaultImpl(t, key, default: untyped): untyped =
+  if key in t: t[key] else: default
```

Comment:
This needs to be optimized so that the hashing is only done once, like the other `getOrDefaultImpl` above this one. E.g something like this:
```nim
  mixin rawGet
  var hc: Hash
  var index = rawGet(t, key, hc)
  if index >= 0: result = t.data[index].val else: default
```